### PR TITLE
Fixed "Message being edited is cancelled on pressing any key"…

### PR DESCRIPTION
**Pull request by kursat**
While editing a message, setting input to empty string was causing edit box disappear.

### Details

1. `ReportActionItemMessageEdit` component is rendering when draft is not an empty string. So we saved original message as a draft when user delete everything from the textarea. This fixed main problem. 
2. `debouncedSaveDraft` was not getting last changes. Changed function to use its parameter, removed last parameter of binding of `_.debounce` to `debouncedSaveDraft`. So it takes last changes now.
3. Cancelled `debouncedSaveDraft`, it was causing another bug when you type and press enter in 1s (debounce duration).

### Fixed Issues
$ https://github.com/Expensify/App/issues/6155

### Tests | QA Steps

1. Right click on a comment.
2. Click edit comment.
3. Remove all text.
4. Wait more than 1 second.
5. Be sure textarea is not disappearing.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Android

#### Web
https://user-images.githubusercontent.com/1279379/142645826-462e0209-06de-471f-8c12-a0b519c8266c.mp4

#### Mobile Web
https://user-images.githubusercontent.com/1279379/142731426-53035901-4299-4668-bfe6-f5e473b99b24.mp4

#### Android
https://user-images.githubusercontent.com/1279379/142730254-1e5efcd2-dd23-4257-a582-ee0ba03deaf8.mp4

### CODE REVIEW

**Comment by kursat** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/pages/home/report/ReportActionItemMessageEdit.js
Comment (tgolen) [inline src/pages/home/report/ReportActionItemMessageEdit.js:85]: Please add a code comment explaining why the two different values are passed to `debouncedSaveDraft` based on the draft length. It will help future developers that come across this to understand the intention.

File: src/pages/home/report/ReportActionItemMessageEdit.js
Comment (tgolen) [inline src/pages/home/report/ReportActionItemMessageEdit.js:115]: Please add a comment explaining why this cancel needs to be here.

**Review by tgolen — CHANGES_REQUESTED** [review]
Looks good, just requesting a few contextual comments.

File: src/pages/home/report/ReportActionItemMessageEdit.js
Comment (kursat) [reply src/pages/home/report/ReportActionItemMessageEdit.js:122]: Hey @tgolen, I've added comments for both. Since English is not my native language, I may have expressed myself poorly. Please let me know if you want to reword anything.

File: src/pages/home/report/ReportActionItemMessageEdit.js
Comment (tgolen) [inline src/pages/home/report/ReportActionItemMessageEdit.js:120]:
```suggestion
         * To prevent re-mount after user saves edit before debounce duration (example: within 1 second), we cancel debounce here.
```

**Review by tgolen — CHANGES_REQUESTED** [review]
Thanks! They look pretty good. I just added one suggestion to make it a little more clear.

Can you also please do all the comments as inline `//` comments? We usually only do block `/**/` comments for method docs or proptype docs

**Review by parasharrajat — CHANGES_REQUESTED** [review]
Good work @kursat. 
Few requests:
1. Please test it on all platforms and Add a video for all of them which were mentioned on the template.
2. Add Tests and QA steps, we can't move forward without those.

**Comment by kursat** [comment]
Thanks @parasharrajat. I'll add test and q/a steps asap. 

About the videos, I recorded one for android and I can record one for mobile web but since I don't have access a mac I'll not be able to record ios and desktop videos. Will be this 3 videos enough?

**Comment by parasharrajat** [comment]
Thanks @kursat .

For other question, it's not my call. Although I think it's fine.

**Comment by kursat** [comment]
Thanks for the enlightenment @parasharrajat, @tgolen is this ok?

> Thanks @parasharrajat. I'll add test and q/a steps asap.
> 
> About the videos, I recorded one for android and I can record one for mobile web but since I don't have access a mac I'll not be able to record ios and desktop videos. Will be this 3 videos enough?

**Comment by kursat** [comment]
Hey @tgolen, this PR is not merged for a while, is there something wrong?

**Review by tgolen — APPROVED** [review]
Sorry for the delay while I was out on vacation.

I don't think we've come to a good conclusion about developers that aren't able to build iOS or desktop. I think in this case, it seems like a pretty safe change that shouldn't have too much platform specific problems, so I think it's OK. 

In other cases where there might be something platform specific, reaching out to the slack channel to see if someone can test out the branch and record it for you might be a good route to try.

**Comment by kursat** [comment]
Hey, no problem and sorry to bother you. I hope you had a good vacation :)  

It's my first PR, and I am trying to figure out how things work. So I am a little nervous. Should I do something to merge this PR? 

That's a great idea, I'll do that next time.
> In other cases where there might be something platform specific, reaching out to the slack channel to see if someone can test out the branch and record it for you might be a good route to try.

**Comment by tgolen** [comment]
> Should I do something to merge this PR?

Ah, no. That's my bad. I was waiting for @parasharrajat to approve because I saw he did a partial review and when there are multiple reviewers, it's good to wait until everyone has approved. However, I think Rajat was really just doing a process-check and not really a code review, so I will merge this.

**Pull request merged**
